source/shared/core_results.cpp

```diff
@@ -1563,6 +1563,19 @@ SQLPOINTER read_lob_field( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT field_in
 
         SQLSRV_ASSERT( SQL_SUCCEEDED( r ), "Unknown SQL error not triggered" );
 
+        if ( stmt->conn->ce_option.enabled == true ) {
+            // cursor type SQLSRV_CURSOR_BUFFERED has to be FORWARD_ONLY
+            // thus has to close and reopen cursor to reset the cursor buffer
+            core::SQLCloseCursor(stmt);
+            core::SQLExecute(stmt);
+            // FETCH_NEXT until the cursor reaches the row that it was at
+            for (int i = 0; i <= row_count; i++) {
+                core::SQLFetchScroll(stmt, SQL_FETCH_NEXT, 0);
+            }
```

Comment:
I've double checked. I get an error "Protocol error in TDS stream" when trying to fetch an encrypted max type column with non FORWARD_ONLY cursor (CURSOR_STATIC AND CURSOR_DYNAMIC). https://github.com/Microsoft/msphpsql/blob/dev/source/shared/core_stmt.cpp#L1126 is where the driver sets the cursor to SQL_CURSOR_FORWARD_ONLY when SQLSRV_CURSOR_BUFFERED is on.